curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/64ae550d9fdb40025e5857ae.md

```diff
@@ -0,0 +1,367 @@
+---
+id: 64ae550d9fdb40025e5857ae
+title: Step 47
+challengeType: 0
+dashedName: step-47
+---
+
+# --description--
+
+You need to run the newly created `straightDetector` function to see the results on the display. To do this, create a function named `findRollResult` and set the argument to `arr`. Then, call the function `straightDetector` with the argument of `arr`.
+
+# --hints--
+
+You should create a function named `findRollResult` with an argument of `arr`.
+
+```js
+assert.match(code, /function\s*findRollResult\s*\(\s*arr\s*\)\s*{/);
```

Comment:
Doesn't pass for arrow function, here is the updated regex.

```suggestion
assert.match(code, /const\s*findRollResult\s*=\s*\(arr\)\s*=>\s*{\s*}/);
```

Updated regex passes for:

```js
const findRollResult = (arr) => {
};
```

Reply:
writing this code will accept both a regular and a es6 arrow function
```/(const\s*findRollResult\s*=\s*\(arr\)\s*=>\s*{\s*}|function\s*findRollResult\s*\(arr\)\s*{\s*})/```